### ISSUE TITLE
Bypassing high ride limitation is possible

### ISSUE THREAD

**Issue by ccoors** [OpenRCT2/OpenRCT2#4965]
**OS:** Arch Linux
**Version:** 0.0.6
**Commit/Build:** 99a662f

In a park with limitation of high rides, one can bypass this limitation by lowering the ground after the ride has been built. I think this is not intended, but it was also possible in the original RCT2.

- [x] Reproducible in RCT2 (vanilla)?
- [ ] Multiplayer?

**Steps to reproduce:**
1. Start a new game in a park with height limitation (Asia - Japanese Coastal Reclaim/Okinawa Coast)
2. Lower the ground under a track piece of Bullet Coaster 1 as far as possible
3. Close Bullet Coaster 1 and delete that track piece
4. Try to rebuild that piece, it is not possible

**Comment by Gymnasiast** [comment]
It will work, but it's expensive. And used on a small scale, it might help to ease the frustration this kind of scenarios cause. So I'm not really sure if this is should be fixed or not.

**Comment by IntelOrca** [comment]
Its sounds like a bug that should be fixed.

**Comment by e-foley** [comment]
Yeah... This is a tricky matter to decide on.  RCT1's Rainbow Valley didn't have this issue because the scenario prohibited landscape changes in addition to tall rides, but RCT2's Rainbow Summit didn't impose the landscape restriction.  (In any case, the two rules can be selected independently in RCT2's scenario editor.)

I think IntelOrca's proposal makes sense.  The idea of manipulating single-tile-wide lines of land to dodge height limits is probably not in the spirit of the scenarios, but we ought to handle the mechanic in a way that makes

**Comment by Christipher** [comment]
Yeah, and, while we are on that issue, doesn't 35ft. (height limit of 7 over the ground) seem a bit low (the Japanese Coastal Reclaim scenario actually exhibits this bug from the start with one of the roller coasters being around 45 -50 ft over the ground)?
